### PR TITLE
feat(add linter): Add flutter_lints linter

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
### Problem

The current linter on [ardrive-web](https://github.com/ardriveapp/ardrive-web) is the [pedantic](https://pub.dev/packages/pedantic). It was deprecated and its rules are not up-to-date with the best practices that the dart team recommends.

### Solution

The best option for today is the [flutter_lints](https://pub.dev/packages/flutter_lints) package. It is maintained by the Flutter team and has a very good set of rules. This package is built on top of Dart's recommended.yaml set of lints from [package:lints](https://pub.dev/packages/lints).

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 2
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.dart]
+indent_brace_style = K&R
```
Comment (fedellen) [inline .editorconfig:12]: nit: missing line at end of file

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 2
+indent_style = space
```
Comment (fedellen) [inline .editorconfig:7]: I don't see spaces as indent style much. Is that how the codebase is setup already?

**Review by fedellen — DISMISSED** [review]
🙌

**Review by karlprieb — APPROVED** [review]
LGTM, I think we just need to add a line at end of file as @fedellen commented.

File: .editorconfig (hunk 1 above)
Comment (thiagocarvalhodev) [reply .editorconfig]: [4f22a59](https://github.com/ardriveapp/ardrive-web/pull/482/commits/4f22a590e84fe2a17f3ee0d07322094d047acccd)

**Review by fedellen — DISMISSED** [review]

**Review by fedellen — APPROVED** [review]

**Pull request merged**